openpype/hosts/maya/plugins/load/load_reference.py

```diff
@@ -36,7 +36,7 @@ class ReferenceLoader(openpype.hosts.maya.api.plugin.ReferenceLoader):
     # Name of creator class that will be used to create animation instance
     animation_creator_name = "CreateAnimation"
 
-    def process_reference(self, context, name, namespace, options):
+    def process_reference(self, context, name, namespace, options, group_name=None):  # noqa
```

Comment:
If we're changing the signature of this method. Shouldn't we be doing so on the base class too? Plus `load_look` would need to be adjusted too? (Even thought it doesn't generate a group?) There's also a `_load_animation.py` which likely needs to be refactored too?

Those would likely get called from the base class too, no?
This PR might currently be breaking those loaders? (not entirely sure)

Reply:
I don't think this is a problem since the process_reference method is to be implemented by subclasses. If you look at the base class, it is clearly said in `openpype/hosts/maya/api/plugin.py` lines 212 to 214:
```
def process_reference(self, context, name, namespace, data):
    """To be implemented by subclass"""
    raise NotImplementedError("Must be implemented by subclass")
```

Reply:
Yes. But the method signature shouldn't change. If we are changing the signature all child classes will need to follow.